src/main/java/sg/ncl/MainController.java

```diff
@@ -100,6 +103,7 @@
     private static final String CONNECTION_ERROR = "Connection Error";
     private final String permissionDeniedMessage = "Permission denied. If the error persists, please contact " + CONTACT_EMAIL;
     private static final String ERR_START_DATE_AFTER_END_DATE = "End date must be after start date";
+    private static final String ERR_INVALID_EMAIL_PASSWORD = "Login failed: Invalid email/password.";
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') 'PASSWORD' detected in this expression, review this potentially hardcoded credential. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2068)